[tools/depends] native and configure fixes

fuzzard opened the pull request:
## Description
Fixes some old variable usage (zlib)
Introduce usage of CMAKE_CXX_COMPILER_LAUNCHER for flatbuffer usage of ccache
assorted other fixes/changes to support Apple Silicon build host

## Motivation and context
Several fixes/cleanups to enable Apple silicon build host.

## How has this been tested?
Successful native dependency build on Apple Silicon host

## What is the effect on users?
N/A

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
[tools/depends][configure.ac] darwin set arch flag for host_includes

explicitly set arch flags for detected host arch type

Commit:
[tools/depends] configure.ac introduce BUILD_CPU variable

Apple silicon now provides a valid use case for a non x86_64 build host.
introduce BUILD_CPU variable to allow testing of native build host based on CPU arch

Commit:
[tools/depends] configure.ac introduce ccache variable

cmake 3.4 introduced CMAKE_CXX_COMPILER_LAUNCHER
flatbuffers cannot handle Compiler set to "ccache compiler". use launcher to enable
ccache usage.

Commit:
[tools/depends] configure.ac add option for clang for native compiler

Currently with macos, gcc/g++ are symlinks/forwarders onto clang/clang++
Enable us to just use clang directly.

Commit:
[tools/depends][native] yasm provide flags to yasm varaibles

Apple silicon native trys to build as x86_64 without this.

Commit:
[tools/depends][native] zlib change variable to match current usage

BUILD_CFLAGS was removed a while ago, and wasnt replaced with its counterpart NATIVE_CFLAGS

Commit:
[tools/depends] configure.ac move darwin to use xcrun instead of xcodebuild

Look to enable a CLT only based build system by using xcrun instead of xcodebuild

Commit:
[tools/depends][target] openssl darwin cpu type fix

garbear reviewed (approved):
ccache support for flatbuffers is a welcome change.

Commit:
[cmake] use modern CMAKE_XXX_COMPILER_LAUNCHER instead of global RULE_LAUNCH_COMPILE

Commit:
[tools/depends][native] libjpeg-turbo be explicit with build flags

Commit:
[tools/depends][target] fribidi cleanup exports

export the "correct" used flag options

Commit:
[tools/depends][target] gmp fix missing build flag

apply cflags to CC so autoconf tests succeed

Commit:
[tools/depends][target] python3 fix cross compile from apple silicon build host

Add patch to default autoconf test to set ac_osx_32bit to no for cross compile.
All darwin platforms we support are 64bit only.

fuzzard commented:
After a bit of work, have green across the board.
With this changeset, native depends builds on Apple silicon, and I can cross compile an x86_64 build that runs in Rosetta.

No more commits for this patch set, I'll wait for this to get approval and then go back to working through apple silicon target build

kambala-decapitator commented on tools/depends/configure.ac:
```diff
@@ -171,21 +172,21 @@ case $build in
     host_includes+=" -isysroot $native_sdk_path"
 
-    android_toolchain_name="darwin-x86_64"
+    CC_FOR_BUILD=[`$use_xcrun -find clang`]
```
should be either `-f` or `--find` per `xcrun`'s manual, there's no `-find` (single dash version)

fuzzard replied:
Odd, it seems to work on Monterey. I'll go back and have a look at it

fuzzard replied:
single dash operators all worked. Ive changed all to match double dash though just for uniformity of documented man

fuzzard commented on cmake/modules/FindCCache.cmake:
```diff
@@ -13,8 +13,8 @@ find_package_handle_standard_args(CCache REQUIRED_VARS CCACHE_PROGRAM)
   # Supports Unix Makefiles, Ninja and Xcode
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${CCACHE_PROGRAM}")
-  set_property(GLOBAL PROPERTY RULE_LAUNCH_LINK "${CCACHE_PROGRAM}")
+  set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM}" CACHE STRING "" FORCE)
+  set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM}" CACHE STRING "" FORCE)
```
@wsnipex Is this correct, in particular the caching?

I found without the CACHE STRING "" FORCE that the launcher items werent listed in cmakecache.txt. 
Because apple uses wrapper scripts, i havent been able to test which is appropriate

wsnipex replied:
yes, that looks fine, we'd want that in CMakeCache.txt
